source/marsGameHud.vwf.yaml

```diff
@@ -0,0 +1,468 @@
+---
+extends: http://vwf.example.com/hud/overlay.vwf
+implements:
+  - http://vwf.example.com/sceneGetter.vwf
+properties:
+  visible: true
+  lastBlocklyNodeID:
+methods:
+  elementPreDraw:
+    parameters:
+      - context
+      - element
+    body: |
+      var alpha = 1;
+      var time = Date.now() / 1000;
+      if ( !element.enabled ) {
+        alpha = 0.5;
+      } else if ( element.isBlinking ) {
+        if ( time  - element.lastBlinkTime > element.blinkInterval ) {
+          alpha = 0.5;
+          if ( time - element.lastBlinkTime > element.blinkInterval + element.blinkDuration ) {
+            element.lastBlinkTime = time;
+          }
+        }
+      }
+      context.globalAlpha = alpha;
```

Comment:
Is there a reason to have the code in yaml?  Wherever possible, I'd rather have it in a .js file...